pkg/pillar/cmd/volumemgr/handlework.go

```diff
@@ -235,8 +235,13 @@ func casIngestWorker(ctxPtr interface{}, w worker.Work) worker.WorkResult {
 		}
 	}
 
+	appImgName := status.ReferenceID()
+	if ctx.hvTypeKube && status.IsContainer() {
+		appImgName = types.KubeContainerImagePrefix + appImgName
+	}
```

Comment:
You've changed all but two of the callers to ReferenceID() to add this prefix, but  the conditions are not identical - some check for additional things. I don't know if that is intentional or accidental.

Wouldn't it make sense to instead change ReferenceID() to do this in one place?
If so then ctx.hvTypeKube needs to be either an argument to ReferenceID, or set as a new boolean in the ContentTreeStatus when it is created.
That level of uniformity/consistency and information hiding would make this easier to understand and maintain.
2)